### PR TITLE
TargetOS::IsMacOS was not properly handled on X64

### PR DESCRIPTION
**Pull request by davidwrighton**
- Add a new concept of only determining the OS dynamically on Unix, and not all the various flags
- Funnel that through our build system
- Remove some legacy msbuild logic that is no longer used

### CODE REVIEW

**Comment by davidwrighton** [comment]
@kunalspathak Does this fix the issue you were seeing?

**Comment by kunalspathak** [comment]
> @kunalspathak Does this fix the issue you were seeing?

I verified that it fixes the problem.

File: src/coreclr/clrdefinitions.cmake [hunk 1]
```diff
@@ -256,6 +256,9 @@ function(set_target_definitions_to_custom_os_and_arch)
       target_compile_definitions(${TARGETDETAILS_TARGET} PRIVATE TARGET_APPLE)
       target_compile_definitions(${TARGETDETAILS_TARGET} PRIVATE TARGET_OSX)
     endif()
+    if (TARGETDETAILS_OS STREQUAL "unix_anyos")
```
Comment (kunalspathak) [inline src/coreclr/clrdefinitions.cmake:259]: what exactly is `unix_anyos`? it represents both linux and osx? because there is also `unix_osx`?

Comment (kunalspathak) [reply src/coreclr/clrdefinitions.cmake:259]: ah, its similar to how we have `universal` term...this is cross plat binary that can be used for altjit purpose on unix or macos I assume.

Comment (davidwrighton) [reply src/coreclr/clrdefinitions.cmake:259]: Yes. Its not actually intended for altjit (although it does work for that case) its primary use is for crossgen2, which supports targeting all architecture/os pairs from any given 64bit build.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**